x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/InferencePyTorchAction.java

```diff
@@ -84,12 +84,13 @@ protected void doRun() throws Exception {
             logger.debug(() -> format("[%s] skipping inference on request [%s] as it has timed out", getDeploymentId(), getRequestId()));
             return;
         }
+        final String requestIdStr = String.valueOf(getRequestId());
         if (isCancelled()) {
+            getProcessContext().getTimeoutCount().incrementAndGet();
```

Comment:
The `onTimeout()` function protects these 2 lines with a check on `if (notified.compareAndSet(false, true)) {` to prevent the timeout count being incremented twice. Perhaps just call `onTimeout` here and optionally pass a different error message.